### DISCUSSION TITLE
💖 vagrant plugin welcomes new contributors and maintainers

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible-community/molecule-plugins#203] [vagrant]
I am unable to maintain this driver as I am not using it, so I am looking for maintainers. 

I can keep providing support for integration with molecule itself but not around vagrant bits.

I would work noting that the top priority would be to setup CI jobs.

**Comment by hswong3i** [comment]
@ssbarnea could i give a hand for this driver?

I am now using it for all of my Ansible roles, in Travis CI + Libvirt + KVM, see:
* https://travis-ci.org/alvistack/ansible-role-bootstrap/jobs/649746955
* https://github.com/alvistack/ansible-role-bootstrap/blob/7f45e943c34ec8a170c145b662bda21c9ac4c9e4/.travis.yml
* https://github.com/alvistack/ansible-role-bootstrap/blob/7f45e943c34ec8a170c145b662bda21c9ac4c9e4/molecule/ubuntu-18.04/molecule.yml

**Comment by jonashackt** [comment]
I also would say CI jobs would be a show stopper for anyone trying to help with this (I also think about helping out...). I was already loosely involved in https://github.com/ansible-community/molecule/pull/1600#issuecomment-445828834 and tried to create an example pipeline using a free CI platform to be able to create VagrantBoxes (with VirtualBox) in https://github.com/jonashackt/vagrant-ansible-on-appveyor - but sadly failed back then.

Now with the Molecule v3 release I stepped back into the game, since there was some development at AppVeyor since then and there seems to be a possiblity right now, to use Vagrant with them. Therefor I filed an issue https://help.appveyor.com/discussions/questions/46774-nested-virtualization-needed-for-opensource-projects-vt-x-is-not-available-verr_vmx_no_vmx - let's see, if they will help here. Will come back, if there's any progress. I don't really see other free options aside from AppVeyor, everything else needs to be payed for...

**Comment by jonashackt** [comment]
Ah @hswong3i I misread your solution, sounds great. Will look into it.

**Comment by jonashackt** [comment]
Wow @hswong3i this really works... Vagrant + libvrt Plugin + libvrt + KVM on Travis... nice - full example here https://github.com/jonashackt/vagrant-travisci-libvrt. So this could be also a way of creating the CI jobs for Molecule 👍

**Comment by sio** [comment]
@jonashackt, CirrusCI also offers free virtualization: https://cirrus-ci.org/guide/linux/#kvm-enabled-privileged-containers

I'm testing my Ansible roles there.

**Comment by ssbarnea** [comment]
I welcome anyone to join this. You will need to document for each CI how to setup Vagrant and assure that it is working. Installing vagrant does not mean it will be able to start a VM.

I am currently working on fixing this for Zuul CI, which is what we use for testing.

**Comment by ssbarnea** [comment]
CI is working, feel free to start creating PR now, or reviewing existing ones.

Be sure you use the review option when doing reviews, when approved we should see a gray "v" check. Mentioning because the review option is well hidden in the UI.

**Comment by hswong3i** [comment]
@ssbarnea thank you very much and confirmed after ansible-community/molecule-vagrant#10 get committed with https://github.com/ansible-community/molecule-vagrant/commit/78582f9f7ead498575f0acc12ace7b9e187306ad, no more hacky dirty workaround is required for:
  - `molecule/*/create.yml`
  - `molecule/*/prepare.yml`
  - `molecule/*/destroy.yml`
  - `molecule/*/library/molecule_vagrant.py`

My complete example for migrate from Molecule 2.22.0 to 3.0.2 goes here:
  - https://github.com/alvistack/ansible-role-bootstrap/compare/eb7a0a6a151f82bb5fff7f26737fad03273bc40f...450db6eea91cab03b195fc103f3d3b7b6d345221
  - [requirements.txt](https://github.com/alvistack/ansible-role-bootstrap/blob/450db6eea91cab03b195fc103f3d3b7b6d345221/requirements.txt)
  - [.travis.yml](https://github.com/alvistack/ansible-role-bootstrap/blob/450db6eea91cab03b195fc103f3d3b7b6d345221/.travis.yml)
  - [molecule/ubuntu-18.04/molecule.yml](https://github.com/alvistack/ansible-role-bootstrap/blob/450db6eea91cab03b195fc103f3d3b7b6d345221/molecule/ubuntu-18.04/molecule.yml)

Enjoy ;-)

**Comment by stdevel** [comment]
Thank you so much, @hswong3i - I'm finally able to use Molecule with Vagrant again! 🎉

**Comment by hswong3i** [comment]
@ssbarnea thank you very much for [v0.2](https://github.com/ansible-community/molecule-vagrant/releases/tag/0.2), [example above](https://github.com/ansible-community/molecule-plugins/issues/63) updated, too ;-)

**Comment by hswong3i** [comment]
@jonashackt For Travis CI + Vagrant + VirtualBox we may need `--nested-hw-virt`, which natively enabled for Libvirt + KVM

See https://www.virtualbox.org/manual/ch09.html#nested-virt

**Comment by apatard** [comment]
I've said it on IRC but it's a good idea to say it here too. If help is still needed/wanted, I can try helping in maintaining the vagrant molecule plugin.

**Comment by ssbarnea** [comment]
Please watch the project and do reviews on issues or PRs. Even ping me directly here or on irc, after this I will also give you more privileges.

It would also be very useful if you can find a way to fix the vagrant installation on the broken platforms (so we can remove the voting: false)

**Comment by rgarrigue** [comment]
Thanks @hswong3i for the example. Wish this could be added in the doc.

**Comment by ssbarnea** [comment]
I invited few people but I will keep the ticket open. If you want to help, just add a comment here and mention my name. I will send you an invite so you can start doing triage, reviews.

**Comment by karcaw** [comment]
this code is becoming very important to our team and will be for years I believe, so I'll add my name to the list of people that can help

**Comment by ssbarnea** [comment]
Done, invited all. If I missed anyone mention me in a comment.

Please watch the project for PRs and review them, only when it gets reviews add the "gate" label to tell CI to merge it. There is not manual merge.

**Comment by bbaassssiiee** [comment]
@kikitux is this something for you?  @ssbarnea
